File: javaScript/web/js/maximumSubarray.js
//Two Pointer Array
//let nums = [-2,2,5,-11,6];
//let nums = [1,2,3,4];
let nums = [-1];

let maxSum = maxSubArray(nums);
console.log("ANSWER " + maxSum);


function maxSubArray(nums) {
    var maxSum = -Number.MAX_VALUE;
    let currentSum = 0;

    for (let leftPointer = 0; leftPointer < nums.length; leftPointer++) {
        console.log("Loop Start " + maxSum);
        for (let rightPointer = leftPointer; rightPointer < nums.length; rightPointer++) {
            currentSum = currentSum + nums[rightPointer];
            //console.log(nums[leftPointer] + " " + nums[rightPointer]);
            
            maxSum = Math.max(maxSum, currentSum);
            console.log("Max " + maxSum + " Current " + currentSum);
        }
        //console.log("");
        currentSum = 0;
        
    }
    return maxSum;
}

//console.log(Math.max(Number.MIN_VALUE + 1000, -1));

File: sort/Leetcode/python/leetcode/MaximumSubarray.py
def maxSubArray(nums):
    for element in nums:
        print(element)

    """
    max_sum = 0
    array_length = len(nums)
    left_pointer = 0
    
    while left_pointer < len(nums):
        current_list = [nums[left_pointer]]
        right_pointer = left_pointer + 1

        while right_pointer < array_length:
            current_list.append(nums[right_pointer])
            #print("Left: ", nums[left_pointer], " Right: ", nums[right_pointer])
            right_pointer += 1
        left_pointer += 1
        print(current_list)
    """

#nums = [-2,1,-3,4,-1,2,1,-5,4]
nums = [1,2,3,4]
maxSubArray(nums)


        